mne_bids/read.py

```diff
@@ -166,3 +168,51 @@ def read_raw_bids(bids_fname, bids_root, verbose=True):
         raw.info['bads'] = list(unique_bads)
 
     return raw
+
+
+def fit_trans_from_points(bids_fname, bids_root, verbose):
+    """Produce transformation matrix from MEG and MRI landmark points.
+
+    Will attempt to read the landmarks of Nasion, LPA, and RPA from the sidecar
+    files of (i) the MEG and (ii) the T1 weighted MRI data. The two sets of
+    points will then be used to calculate a transformation matrix from HEAD
+    coordinates to MRI coordinates.
+
+    Parameters
+    ----------
+    bids_fname : str
+        Full name of the data file (not a path)
+    bids_root : str
+        Path to root of the BIDS folder
+    verbose : bool
+        The verbosity level
+
+    Returns
+    -------
+    trans : instance of mne.transforms.Transform
+        The data transformation matrix from HEAD to MRI coordinates
+
+    """
+    # Get the sidecar file for MRI landmarks
+    t1w_json_path = _find_sidecar(bids_root, bids_fname, 't1w.json')
+
+    # Get MRI landmarks from the JSON sidecar
+    with open(t1w_json_path, 'r') as f:
+        t1w_json = json.load(f)
+    mri_coords_dict = t1w_json['AnatomicalLandmarkCoordinates']
+    mri_landmarks = np.asarray((mri_coords_dict['LPA'],
+                                mri_coords_dict['NAS'],
+                                mri_coords_dict['RPA']))
+
+    # Get MEG landmarks from the raw file
+    raw = read_raw_bids(bids_fname, bids_root)
+    meg_coords_dict = _extract_landmarks(raw.info['dig'])
+    meg_landmarks = np.asarray((meg_coords_dict['LPA'],
+                                meg_coords_dict['NAS'],
+                                meg_coords_dict['RPA']))
+
+    # Given the two sets of points, fit the transform
+    trans_fitted = fit_matched_points(src_pts=meg_landmarks,
+                                      tgt_pts=mri_landmarks)
```

Comment:
This will work assuming that the "MEG landmarks" and "MRI landmarks" are in the same coordinate space. However, this is not the case. The fiducials are in RAS coordinate system once you do apply the trans but the "MRI landmarks" are in "voxel" coordinate space. You can get the transformation to go from voxel to RAS in the nifti header. That's why you need `vox2ras` that @agramfort hinted towards. 

Also only Elekta systems are in RAS coordinates, other systems have different coordinate frames. There is a field in BIDS which tells you which coordinate system is used when storing the fiducials. You can easily convert between them because it's only the orientation and origin which differ. I'm not exactly sure, but the fiducials might already be stored in RAS coordinates when MNE reads them in for other systems (can you check?). 

And with that, I want to officially welcome you to the coordinate system rabbit hole. May the force be with you!

Reply:
You can use nilearn to verify visually that what you get is correct: http://nilearn.github.io/modules/generated/nilearn.plotting.plot_anat.html#nilearn.plotting.plot_anat

Reply:
I actually need `ras2vox` as well ... because it all starts with writing the MRI in BIDS format:

- I get a `raw` with fiducials and a `trans` to put these fiducials from head coords to MRI coords
- I apply `trans` to my fiducials ... now they are in MRI coords ... but the same space as `raw` (=neuromag RAS)

... so then I need to apply `ras2vox` to get these coords into `voxel` space for the particular `*_T1w.nii.gz` MRI that I have --> and then save these fiducials (in `voxel` units) to the `*_T1w.json` sidecar


... later, (for reading `trans`, i.e., fitting the sets of points):

- I read my neuromag RAS coords of fiducials from `raw` 
- I read `voxel` space fiducials from `*_T1w.json`
- I transform these `voxel` space fiducials to Neuromag RAS

sounds quite complicated :( I am still looking at nibabel and freesurfer

Reply:
see a first attempt [here](https://gitter.im/mne-tools/mne-gsoc-2019-BIDS?at=5d19f1e3189e7b4ff04838d5)

Reply:
yes, this sounds about right what you describe. Now you need to make sure it's all "correct" by visualizing intermediate steps.

Reply:
yep, see my comment https://github.com/mne-tools/mne-bids/pull/211#issuecomment-507339263